Tests/NIOTests/CodecTest.swift

```diff
@@ -210,19 +210,43 @@ public final class ByteToMessageDecoderTest: XCTestCase {
         _ = try channel.pipeline.addHandler(decoder).wait()
 
         // We're going to send in 513 bytes. This will cause a chunk to be passed on, and will leave
-        // a 512-byte empty region in a 513 byte buffer. This will not cause a shrink.
+        // a 512-byte empty region in a byte buffer with a capacity of 1024 bytes. Since 512 empty
```

Comment:
I think we should leave the old unit tests alone and add targeted new tests that specifically test for what was broken.

Reply:
With the fixed logic above this tests fails without modifications. For this reason, I adjusted it.

Reply:
Thank you! That makes sense.